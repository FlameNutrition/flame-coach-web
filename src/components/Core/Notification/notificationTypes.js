export const INFO = 'INFO';
export const WARNING = 'WARNING';
export const ERROR = 'ERROR';
export const SUCCESS = 'SUCCESS';
