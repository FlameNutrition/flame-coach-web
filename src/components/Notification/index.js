import React from 'react';
import {
  Box, Collapse, IconButton, makeStyles
} from '@material-ui/core';
import { Alert } from '@material-ui/lab';
import CloseIcon from '@material-ui/icons/Close';
import PropTypes from 'prop-types';
import clsx from 'clsx';

const useStyles = makeStyles(() => ({
  root: {},
  notificationBar: {
    marginTop: '10px'
  }
}));

const Notification = ({
  open,
  openHandler,
  message,
  level,
  collapse,
  className
}) => {
  const classes = useStyles();

  const NOTIFICATION_ERROR = 'error';
  const NOTIFICATION_WARNING = 'warning';
  const NOTIFICATION_INFO = 'info';
  const NOTIFICATION_SUCCESS = 'success';

  let internalLevel;

  switch (level) {
    case 'INFO':
      internalLevel = NOTIFICATION_INFO;
      break;
    case 'WARNING':
      internalLevel = NOTIFICATION_WARNING;
      break;
    case 'ERROR':
      internalLevel = NOTIFICATION_ERROR;
      break;
    case 'SUCCESS':
      internalLevel = NOTIFICATION_SUCCESS;
      break;
    default:
      internalLevel = NOTIFICATION_INFO;
      break;
  }

  return (
    <Box className={clsx(className, classes.notificationBar)}>
      {collapse
        ? (
          <Collapse in={open}>
            <Alert
              variant="filled"
              severity={internalLevel}
              action={(
                <IconButton
                  aria-label="close"
                  color="inherit"
                  size="small"
                  onClick={() => {
                    openHandler(false);
                  }}
                >
                  <CloseIcon fontSize="inherit" />
                </IconButton>
              )}
            >
              {message}
            </Alert>
          </Collapse>
        ) : (
          <Alert
            itemID="notification"
            variant="filled"
            severity={internalLevel}
          >
            {message}
          </Alert>
        )}
    </Box>
  );
};

Notification.propTypes = {
  message: PropTypes.string.isRequired,
  level: PropTypes.string.isRequired,
  open: PropTypes.bool,
  openHandler: PropTypes.func,
  collapse: PropTypes.bool,
  className: PropTypes.string
};

Notification.defaultProps = {
  collapse: false
};

export default Notification;
