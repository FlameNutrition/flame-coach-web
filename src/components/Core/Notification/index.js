import Box from "@material-ui/core/Box";
import Collapse from "@material-ui/core/Collapse";
import IconButton from "@material-ui/core/IconButton";
import { makeStyles } from "@material-ui/core";

import { ERROR, INFO, SUCCESS, WARNING } from "./notificationTypes";

import { Alert } from "@material-ui/lab";
import { Close as CloseIcon } from "@material-ui/icons";
import PropTypes from "prop-types";
import React from "react";
import clsx from "clsx";

const useStyles = makeStyles(() => ({
  root: {},
  notificationBar: {
    marginTop: "10px"
  }
}));

const Notification = ({
  open,
  openHandler,
  message,
  level,
  collapse,
  className
}) => {
  const classes = useStyles();

  let internalLevel;

  switch (level) {
    case INFO:
      internalLevel = "info";
      break;
    case WARNING:
      internalLevel = "warning";
      break;
    case ERROR:
      internalLevel = "error";
      break;
    case SUCCESS:
      internalLevel = "success";
      break;
    default:
      internalLevel = "info";
      break;
  }

  return (
    <Box className={clsx(className, classes.notificationBar)}>
      {collapse
        ? (
          <Collapse in={open}>
            <Alert
              variant="filled"
              severity={internalLevel}
              action={(
                <IconButton
                  aria-label="close"
                  color="inherit"
                  size="small"
                  onClick={() => {
                    openHandler(false);
                  }}
                >
                  <CloseIcon fontSize="inherit" />
                </IconButton>
              )}
            >
              {message}
            </Alert>
          </Collapse>
        ) : (
          <Alert
            itemID="notification"
            variant="filled"
            severity={internalLevel}
          >
            {message}
          </Alert>
        )}
    </Box>
  );
};

Notification.propTypes = {
  message: PropTypes.string.isRequired,
  level: PropTypes.string.isRequired,
  open: PropTypes.bool,
  openHandler: PropTypes.func,
  collapse: PropTypes.bool,
  className: PropTypes.string
};

Notification.defaultProps = {
  collapse: false
};

export default Notification;
