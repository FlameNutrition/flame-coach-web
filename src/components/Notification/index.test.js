import React from 'react';
import toJson from 'enzyme-to-json';
import { createMount, createRender, createShallow } from '@material-ui/core/test-utils';
import sinon from 'sinon';
import Notification from './index';

describe('<Notification/ >', () => {
  let shallow;
  let render;
  let mount;

  beforeAll(() => {
    shallow = createShallow();
    render = createRender();
    mount = createMount();
  });

  it('Show info notification', () => {
    const notification = shallow(
      <Notification message="INFO MESSAGE" level="INFO" />
    );

    expect(toJson(notification)).toMatchSnapshot();
  });

  it('Show warning notification', () => {
    const notification = shallow(
      <Notification message="INFO MESSAGE" level="WARNING" />
    );

    expect(toJson(notification)).toMatchSnapshot();
  });

  it('Show error notification', () => {
    const notification = shallow(
      <Notification message="INFO MESSAGE" level="ERROR" />
    );

    expect(toJson(notification)).toMatchSnapshot();
  });

  it('Show success notification', () => {
    const notification = shallow(
      <Notification message="INFO MESSAGE" level="SUCCESS" />
    );

    expect(toJson(notification)).toMatchSnapshot();
  });

  it('Show notification with collapse', () => {
    const notification = shallow(
      <Notification
        collapse
        message="INFO MESSAGE"
        level="SUCCESS"
      />
    );

    expect(toJson(notification)).toMatchSnapshot();
  });

  it('Message should be "I like you"', () => {
    const notification = shallow(
      <Notification message="I like you" level="SUCCESS" />
    );

    const message = notification.find({ itemID: 'notification' })
      .children()
      .text();

    expect(message).toEqual('I like you');
  });

  it('Close notification', () => {
    const openHandlerMock = sinon.spy();
    const notification = mount(
      <Notification
        collapse
        open
        message="INFO MESSAGE"
        level="SUCCESS"
        openHandler={openHandlerMock}
      />
    );

    const closedBtn = notification.find({ type: 'button' });

    closedBtn.simulate('click');

    expect(openHandlerMock).toHaveBeenCalledOnce();
  });
});
