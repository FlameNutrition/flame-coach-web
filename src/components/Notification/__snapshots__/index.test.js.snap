// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification/ > Show error notification 1`] = `
<Styled(MuiBox)
  className="makeStyles-notificationBar-2"
>
  <WithStyles(ForwardRef(Alert))
    itemID="notification"
    severity="error"
    variant="filled"
  >
    INFO MESSAGE
  </WithStyles(ForwardRef(Alert))>
</Styled(MuiBox)>
`;

exports[`<Notification/ > Show info notification 1`] = `
<Styled(MuiBox)
  className="makeStyles-notificationBar-2"
>
  <WithStyles(ForwardRef(Alert))
    itemID="notification"
    severity="info"
    variant="filled"
  >
    INFO MESSAGE
  </WithStyles(ForwardRef(Alert))>
</Styled(MuiBox)>
`;

exports[`<Notification/ > Show notification with collapse 1`] = `
<Styled(MuiBox)
  className="makeStyles-notificationBar-2"
>
  <WithStyles(ForwardRef(Collapse))>
    <WithStyles(ForwardRef(Alert))
      action={
        <WithStyles(ForwardRef(IconButton))
          aria-label="close"
          color="inherit"
          onClick={[Function]}
          size="small"
        >
          <Memo
            fontSize="inherit"
          />
        </WithStyles(ForwardRef(IconButton))>
      }
      severity="success"
      variant="filled"
    >
      INFO MESSAGE
    </WithStyles(ForwardRef(Alert))>
  </WithStyles(ForwardRef(Collapse))>
</Styled(MuiBox)>
`;

exports[`<Notification/ > Show success notification 1`] = `
<Styled(MuiBox)
  className="makeStyles-notificationBar-2"
>
  <WithStyles(ForwardRef(Alert))
    itemID="notification"
    severity="success"
    variant="filled"
  >
    INFO MESSAGE
  </WithStyles(ForwardRef(Alert))>
</Styled(MuiBox)>
`;

exports[`<Notification/ > Show warning notification 1`] = `
<Styled(MuiBox)
  className="makeStyles-notificationBar-2"
>
  <WithStyles(ForwardRef(Alert))
    itemID="notification"
    severity="warning"
    variant="filled"
  >
    INFO MESSAGE
  </WithStyles(ForwardRef(Alert))>
</Styled(MuiBox)>
`;
